include/libtwofloat/arithmetics/double-word-arithmetic.hpp

```diff
@@ -233,5 +233,628 @@ inline two<T> div(const two<T> &x, const two<T> &y) {
+  {8.314696123025452357e-01, 1.407385698472802389e-18},
+  {7.071067811865475727e-01, -4.833646656726456726e-17}
+};
+static const double fp64_sin_table [4][2] = {
+  {1.950903220161282758e-01, -7.991079068461731263e-18},
+  {3.826834323650897818e-01, -1.005077269646158761e-17},
+  {5.555702330196021776e-01,  4.709410940561676821e-17},
+  {7.071067811865475727e-01, -4.833646656726456726e-17}
+};
+
+// Notice the definition of the qd namespace
+namespace qd {
+
+  // Reference: QD / inline.h
+  /* Computes the nearest integer to input. */
+  template <typename T>
+  inline T nint(T input) {
+    T pointfive;
+
+    if constexpr (std::is_same_v<T, float>) {
+      pointfive = 0.5f;
+    } else if constexpr (std::is_same_v<T, double>) {
+      pointfive = 0.5;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
+
+    if(input = std::floor(input)) {
+      return input;
+    }
+    return std::floor(input + pointfive);
+  }
+
+  // TODO: already implemented in twofloat?
+  // Reference: QD / inline.h
+  /* Computes fl(a+b) and err(a+b). Assumes |a| >= |b|. */
+  template <typename T>
+  inline T quick_two_sum(T a, T b, T &err) {
+    T s = a + b;
+    err = b - (s - a);
+    return s;
+  }
+
+  // TODO: make sure it has not been already implemented in twofloat
+  // Reference: QD / inline.h
+  /* Computes high word and low word of input */
+  template <typename T>
+  inline void split(T input, T &hi, T &lo) {
+    T temp;
+
+    T local_qd_splitter;
+    T local_qd_split_thresh;
+    T local_split_factor;
+    T local_split_factor_2;
+    if constexpr (std::is_same_v<T, float>) {
+      local_qd_splitter = fp32_qd_splitter;
+      local_qd_split_thresh = fp32_qd_split_thresh;
+      local_split_factor = fp32_split_factor;
+      local_split_factor_2 = fp32_split_factor_2;
+    } else if constexpr (std::is_same_v<T, double>) {
+      local_qd_splitter = fp64_qd_splitter;
+      local_qd_split_thresh = fp64_qd_split_thresh;
+      local_split_factor = fp64_split_factor;
+      local_split_factor_2 = fp64_split_factor_2;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
+
+    if(input > local_qd_split_thresh || input < -local_qd_split_thresh) {
+      input *= local_split_factor;
+      temp = local_qd_splitter * input;
+      hi = temp - (temp - input);
+      lo = input - hi;
+      hi *= local_split_factor_2;
+      lo *= local_split_factor_2;
+    } else {
+      temp = local_qd_splitter * input;
+      hi = temp - (temp - input);
+      lo = input - hi;
+    }
+  }
+
+  // Reference: QD / inline.h
+  /* Computes fl(input * input) and err(input * input) */
+  template <typename T>
+  inline T two_sqr(T input, T &err) {
+
+    T twopointzero;
+    if constexpr (std::is_same_v<T, float>) {
+      twopointzero = 2.0f;
+    } else if constexpr (std::is_same_v<T, double>) {
+      twopointzero = 2.0;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
```

Comment:
Just assign 2.0 here or even write something link:
`err = ((hi * hi - q) +(T)2.0 * hi * lo) + lo * lo;`

Reply:
Addressed in [800b777](https://github.com/esa-tu-darmstadt/twofloat/pull/1/commits/800b777d24967acc2369cbff64d38228cbf07af9).